File: 0325/SoohyunKim/1106_호텔.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
// 고객 수, 도시 개수, 도시홍보[[비용, 고객]]
const [[C, N], ...cities] = input.map(v => v.split(' ').map(Number));

const MAX_C = C + 100; // 고객 <= 100
const dp = new Array(MAX_C).fill(Infinity);
dp[0] = 0;

for (let i = 0; i < N; i++) {
  const [cost, customers] = cities[i];
  
  // 고객 배수 돌면서 dp 갱신
  for (let j = customers; j < MAX_C; j++) { 
    dp[j] = Math.min(dp[j], dp[j - customers] + cost);
  }
}

let answer = Infinity;
for (let i = C; i < MAX_C; i++) {
  answer = Math.min(answer, dp[i]);
}

console.log(answer);

File: 0325/SeoyeonHong/1106 호텔.py
# 호텔의 고객을 적어도 C명 늘이기 위해 형택이가 투자해야 하는 돈의 최솟값
import sys

input = sys.stdin.readline
C, N = map(int, input().split())
ad = []
INF = int(1e9)
dp = [INF] * (C+100)
dp[0] = 0

for _ in range(N):
    ad.append(list(map(int, input().split())))
    
for cost, customer in ad:
    for i in range(customer, C+100):
        dp[i] = min(dp[i-customer]+cost, dp[i])

print(min(dp[C:]))



